integration-cli/docker_api_inspect_test.go

```diff
@@ -32,20 +32,17 @@ func (s *DockerSuite) TestInspectApiContainerResponse(c *check.C) {
 		body := getInspectBody(c, cs.version, cleanedContainerID)
 
 		var inspectJSON map[string]interface{}
-		if err := json.Unmarshal(body, &inspectJSON); err != nil {
-			c.Fatalf("unable to unmarshal body for version %s: %v", cs.version, err)
-		}
+		err := json.Unmarshal(body, &inspectJSON)
+		c.Assert(err, checker.IsNil, check.Commentf("Unable to unmarshal body for version %s", cs.version))
 
 		for _, key := range cs.keys {
-			if _, ok := inspectJSON[key]; !ok {
-				c.Fatalf("%s does not exist in response for version %s", key, cs.version)
-			}
+			_, ok := inspectJSON[key]
+			c.Assert(ok, checker.True, check.Commentf("%s does not exist in response for version %s", key, cs.version))
```

Comment:
With loops over things to check, I like to use `Check` & `Errorf`, rather than `Assert` and `Fatalf` to let the loop continue to run. That way you get a full report of what fails in a loop vs having to possibly run multiple times to see all the failures. Only a suggestion while we're in here changing code.